utils/dbt_format_utils.py

```diff
@@ -0,0 +1,329 @@
+
+    # Get all dbt Models
+    cte_model_files = glob(f"{path_to_models}/*.sql", recursive=True)
+
+    # Iterate through models and read each one line by line
+    # Then selectively write line to temp file.
+    # Finally, replace file content with temp file content
+    for model_file_path in cte_model_files:
+        skip_line = True
+        with tempfile.TemporaryFile(mode="r+") as temp_file:
+            temp_file.write(config)
+            with open(model_file_path, "r") as model_file:
+                for line in model_file:
+                    # Dont start writing to file if were still in config section
+                    if stop_skip_pattern in line:
+                        skip_line = False
+                    if skip_line:
+                        continue
+                    # Substitute source with 'cta' if regex match
+                    line = re.sub("(?<=source\(').*?(?=', '_airbyte_raw_)", "cta", line)
+                    temp_file.write(line)
+            temp_file.seek(0)
+            with open(f"{model_file_path}", "w") as output_file:
+                output_file.write(temp_file.read())
+
+
+# Function to modify sql files for full-refresh models
+# this will add the logic for partition replacement strat
+def modify_full_refresh_models(path_to_models):
+    # Default Config Section thats gonna be used to replace Airbyte generated config.
+    # This also contains the partition replacement header.
+    partition_replacement_header = (
+        "{% set partitions_to_replace = [\n"
+        '    "timestamp_trunc(current_timestamp, day)",\n'
+        '    "timestamp_trunc(timestamp_sub(current_timestamp, interval 1 day), day)"\n'
+        "] %}\n"
+        "{{ config(\n"
+        '    cluster_by = "_airbyte_emitted_at",\n'
+        '    partition_by = {"field": "_airbyte_emitted_at", "data_type": "timestamp", "granularity": "day"},\n'
+        "    partitions = partitions_to_replace,\n"
+        '    unique_key = "_airbyte_ab_id"\n'
+        ") }}\n"
+    )
+
+    partition_replacement_footer = (
+        "{% if is_incremental() %}\n"
+        'where timestamp_trunc(_airbyte_emitted_at, day) in ({{ partitions_to_replace | join(",") }})\n'
+        "{% endif %}"
+    )
+
+    # Get all full-refresh Models
+    cte_model_files = glob(f"{path_to_models}/*.sql", recursive=True)
+
+    # Iterate through models and read each one line by line
+    # Then selectively write line to temp file.
+    # Finally, replace file content with temp file content
+    for model_file_path in cte_model_files:
+        skip_line = True
+        with tempfile.TemporaryFile(mode="r+") as temp_file:
+            temp_file.write(partition_replacement_header)
+            with open(model_file_path, "r") as model_file:
+                for line in model_file:
+                    # Dont start writing to file if were still in config section
+                    if "-- Final base SQL" in line:
```

Comment:
OK yeah lol so the code I slapped together yesterday is mostly duplicative / not quite as elegant as yours - the one thing I would suggest is adjusting the logic for separating all the config stuff at the top from the SQL model itself. I think a strat that might be more resilient than looking for `-- Final base SQL` would be to treat the config section as a "block" that starts with `{{ config(` and ends with `) }}` - [here's how I twerked that logic in my code.](https://gist.github.com/emily-flambe/caae5b5e873946aab6399bc3d560768e#file-modify_full_refresh_models-py-L60-L64)

Reply:
ooh yeah thats a good idea, Ill switch it to use those strings